source/server/hot_restarting_base.cc

```diff
@@ -0,0 +1,213 @@
+#include "server/hot_restarting_base.h"
+
+#include "common/api/os_sys_calls_impl.h"
+#include "common/common/utility.h"
+
+namespace Envoy {
+namespace Server {
+
+using HotRestartMessage = envoy::HotRestartMessage;
+
+void HotRestartingBase::initDomainSocketAddress(sockaddr_un* address) {
+  memset(address, 0, sizeof(*address));
+  address->sun_family = AF_UNIX;
+}
+
+sockaddr_un HotRestartingBase::createDomainSocketAddress(uint64_t id, const std::string& role) {
+  // Right now we only allow a maximum of 3 concurrent envoy processes to be running. When the third
+  // starts up it will kill the oldest parent.
+  const uint64_t MAX_CONCURRENT_PROCESSES = 3;
+  id = id % MAX_CONCURRENT_PROCESSES;
```

Comment:
Is this incomding id the PID? Or just a small int that rotates 0,1,2,0,1,2,...

Reply:
Yeah, just a sequentially rotating thing. Apparently when first working on this code, I just figured that out from context, because now I can't find any comments describing that. Added.